### PR TITLE
Make `nsobject_prefer_isequal` rule work for nested classes

### PR DESCRIPTION
**Pull request by SimplyDanny**

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift [hunk 1]
```diff
@@ -89,6 +83,14 @@ private extension FunctionDeclSyntax {
     }
 }
 
+private extension Syntax {
+    var isInObjcClass: Bool {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift:87]: maybe the existing rule had this problem already, but we should also check for `@objcmembers`

Comment (SimplyDanny) [reply Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift:87]: You are right. It didn't work before either. Added a CHANGELOG entry for the fix.

File: Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift [hunk 2]
```diff
@@ -89,6 +83,14 @@ private extension FunctionDeclSyntax {
     }
 }
 
+private extension Syntax {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift:86]: I think if you extend SyntaxProtocol you don’t need to wrap with Syntax above.

Comment (SimplyDanny) [reply Source/SwiftLintFramework/Rules/Lint/NSObjectPreferIsEqualRule.swift:86]: Good hint! See #4332.

**Pull request merged**